client/Lykke.Service.Assets.Client/Cache/DictionaryCache.cs

```diff
@@ -1,52 +1,95 @@
-﻿using System;
+﻿using Common;
+using Common.Log;
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Reactive.Disposables;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Lykke.Service.Assets.Client.Cache
 {
+    /// <inheritdoc />
     public class DictionaryCache<T> : IDictionaryCache<T>
         where T : ICacheItem
     {
         private readonly IDateTimeProvider _dateTimeProvider;
-        private readonly TimeSpan          _cacheExpirationPeriod;
+        private readonly TimeSpan _cacheExpirationPeriod;
```

Comment:
I would recommend throw away the whole content of the class and start using a mature solution
https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-2.1
The example can be found here
https://github.com/LykkeCity/Lykke.Service.FeeCalculator/blob/7dbcf7e2c8c25d1b76b0aae345f95795457c843b/client/Lykke.Service.FeeCalculator.Client/FeeCalculatorClientCached.cs#L16

Reply:
There is even a couple of decorators for IMemoryCache on our common library: https://github.com/LykkeCity/CommonDotNetLibraries/tree/master/src/Lykke.Common/Cache
I would recommend to use one of these.